src/utils/docker.js

```diff
@@ -107,7 +107,7 @@ class Docker extends EventEmitter {
                     return Promise.resolve();
                 }
 
-                this.debug && this.logger.error(`Failed to run container: ${ err.message }`);
+                throw err;
```

Comment:
Actual fix! Without an explicit throw error was swallowed and never made it up the promise chain.